include/nudb/error.hpp

```diff
@@ -8,75 +8,12 @@
 #ifndef NUDB_ERROR_HPP
 #define NUDB_ERROR_HPP
 
-#include <boost/system/system_error.hpp>
 #include <boost/system/error_code.hpp>
 
 namespace nudb {
 
-/// The type of system-specific error code returned by the implementation
-#if GENERATING_DOCS
-class error_code{};
-
-#else
-using boost::system::error_code;
-
-#endif
-
-/// The type of cross-platform error code used by the implementation
-#if GENERATING_DOCS
-class error_condition{};
-
-#else
-using boost::system::error_condition;
-
-#endif
-
-/// The type of system-specific exception used when throwing
-#if GENERATING_DOCS
-class system_error{};
-
-#else
-using boost::system::system_error;
-
-#endif
-
-/// Returns the  category used for system-specific error codes
-#if GENERATING_DOCS
-error_category const&
-system_category();
-
-#else
-using boost::system::system_category;
-
-#endif
-
-/// Returns the category used for cross-platform error codes
-#if GENERATING_DOCS
-error_category const&
-generic_category();
-
-#else
-using boost::system::generic_category;
-
-#endif
-
-/// The base class used for error categories
-#if GENERATING_DOCS
-class error_category{};
-
-#else
-using boost::system::error_category;
-
-#endif
-
-/// The set of constants used for cross-platform error codes
-#if GENERATING_DOCS
-enum errc{};
-
-#else
-namespace errc = boost::system::errc;
-
-#endif
+// lift all error code stuff from boost::system into the nudb namespace
+using namespace boost::system; // system_category, errc
```

Comment:
The whole error_code thing through me for a loop.  In large part this is because I wasn't familiar the boost/std libraries.  And of course when I looked to see what they were about: design, purpose, implementation, etc.  It seemed very confusing - only after spending quite a bit of time did I figure it out.

I looked at the documentation in this context.  We didn't want to include a whole explanation of error_code, error_condition, etc. We had our own nudb::error_code - which I thought was a mistake as it is just a typedef.  So I simplified this.  That's how we got here.  As to your specific question: "Do we really want to import the entire namespace"

I generally don't do this as I prefer to use the long name e.g. boost::system::error_code.   I like to be able to see where the stuff is coming from and not risk bringing in something unexpected.  So this time I made an exception.  Why not just import the functions we're using - I didn't think of it.  In fact your syntax : using boost::system::error_category was unfamiliar to me. I might have known it at one time but forgotten it.

So in any case, I agree with you - the following would be better.  I think these are the only things we're actually using.  So I would replace

using namespace boost::system; // system_category, errc

with

using boost::system::error_category;
using boost::system::error_code;